### PR TITLE
15.x, I made the complexity better and made it easier to expand

### PR DESCRIPTION
**Pull request by timjon**

### CODE REVIEW

File: Server/src/usercoordinator.erl [hunk 1]
```diff
@@ -12,59 +12,65 @@ init(Socket) ->
...
-		  %The character ~ is used as the stop character for when the client should stop reading from the tcp connection
-		  Format_result = io_lib:format("~p", [{Class_names, Messages}]) ++ "~",
-	      gen_tcp:send(Port, Format_result),
-	      %Spawns a diagram coordinator for this diagram if it doesnt exist already 
-		  case find_diagram(Did, Diagrams) of 
-	        not_created -> loop(Socket, [{Did, spawn(fun () -> diagramcoordinator:init({Classes, Messages}) end)}| Diagrams])
-	      end
-	  end;
-
 	%This case happens if the connection to the client is lost
```
Comment (pontuslaestadius) [inline Server/src/usercoordinator.erl:20]: Starting all cases with 'this case happens' is a bit redundant, I think the code explains that it is a case pretty clearly already. So my suggestion is to remove those 3 words from all the comments.

File: Server/src/usercoordinator.erl [hunk 2]
```diff
@@ -12,59 +12,65 @@ init(Socket) ->
...
+	  receive
+		ok -> ok
+	  end,
+	  %Receives the result from the diagram coordinator and sends it to the client
+	  loop(Socket, Diagrams)
+  end;
+  
+%This will be selected if its a new diagram selected
+use_input({ok, {Did, Class_names, Classes, Messages}}, Socket, Diagrams) ->
+  %This will be selected if its a new diagram selected
```
Comment (pontuslaestadius) [inline Server/src/usercoordinator.erl]: Duplicate comment

File: Server/src/usercoordinator.erl [hunk 3]
```diff
@@ -12,59 +12,65 @@ init(Socket) ->
...
+	  Pid ! {next_message, self()},
+	  %Confirmation that the diagram coordinator received the message
+	  receive
+		ok -> ok
+	  end,
+	  %Receives the result from the diagram coordinator and sends it to the client
+	  loop(Socket, Diagrams)
+  end;
+  
+%This will be selected if its a new diagram selected
```
Comment (pontuslaestadius) [inline Server/src/usercoordinator.erl]: Don't really understand what this is trying to say.

**Review by pontuslaestadius — CHANGES_REQUESTED** [review]
Needs comment modification

File: Server/src/usercoordinator.erl (hunk 1 above)
Comment (Sebastian-Fransson) [reply Server/src/usercoordinator.erl:20]: But it does add to clarification for future users not too well versed in subject.

File: Server/src/usercoordinator.erl [hunk 4]
```diff
@@ -60,9 +60,8 @@ use_input({ok, {Did, next_message}}, Socket, Diagrams) ->
 	  loop(Socket, Diagrams)
   end;
   
-%This will be selected if its a new diagram selected
+%This patter will match when the first argument is in the format of a new diagram
```
Comment (pontuslaestadius) [inline Server/src/usercoordinator.erl]: pattern?

**Review by pontuslaestadius — APPROVED** [review]

**Review by Sebastian-Fransson — APPROVED** [review]

**Pull request merged**